### ISSUE TITLE
Button to update

### ISSUE THREAD

**Issue by RasmusAntons** [inglor/aarchup#1]
Revival of this issue: https://github.com/aericson/aarchup/issues/12

[Gumik implemented the button](https://github.com/gumik/aarchup) but it never got merged back. I've been using that feature for a long time now and I think it's very convenient. I have also [adapted his changes to the current master](https://github.com/RasmusAntons/aarchup/tree/update-button).

Technically you're supposed to [ask the server about the ability to use actions](https://developer.gnome.org/notification-spec/) first, but since the user has to deliberately use the --update-command option that doesn't seem necessary.

**Comment by inglor** [comment]
I'm not so sure I like this option. It kinda breaks the philosophy of doing one thing only, which is notify you for updates. Also I am thinking Gnome developers weren't probably thinking of this use case when they added callbacks for notifications - it seems _abusing_ the functionality.

A side effect is if you run a virtual console to perform updates then the callback stays active and the timer doesn't start until you shut the window. For me it's common to have virtual consoles windows open for days so I'll be stuck wondering what happen to my notifications.

**Comment by RasmusAntons** [comment]
Yes that's true. The implementation isn't great, it just works for me because I used a non-blocking command (xfce4-terminal).
